### PR TITLE
GS: Add video folder and support additional arguments for video dumping

### PR DESCRIPTION
**Pull request by Tokman5**
### Description of Changes

- Add Recording tab in Graphic settings
- Move video codec, container and bitrate settings into it from Advanced tab
- Add video folder setting
- Add extra arguments setting

### Rationale behind Changes

Now you can record a video in CRF mode while using libx264
### Suggested Testing Steps

download necessary files from #7613 and test video path setting and extra arguments
I don't know much what arguments are supported though

### CODE REVIEW

File: pcsx2/Frontend/HostSettings.cpp [hunk 1]
```diff
@@ -250,5 +251,11 @@ void Host::Internal::UpdateEmuFolders()
 					GSTextureReplacements::ReloadReplacementMap();
 			});
 		}
+
+		if (EmuFolders::Videos != old_videos_directory)
+		{
+			GSEndCapture();
```
Comment (stenzek) [inline pcsx2/Frontend/HostSettings.cpp:257]: This isn't thread-safe. You should do something like:
```suggestion
if (VMManager::HasValidVM())
{
  GetMTGS().RunOnGSThread([]() {
    GSEndCapture();
  });
}
```

**Comment by Tyestor** [comment]
is it possible for you to add the folder to the folders tab?

image

**Comment by Tokman5** [comment]
> is it possible for you to add the folder selection to the folders tab?

it's possible, though it will increase window's height a bit.

**Comment by stenzek** [comment]
You can't have a folder selection in two locations, they'll desynchronize with each other.

In graphics settings is fine, and is the most logical location IMO.

**Comment by Tokman5** [comment]
then i won't move it from Recording tab

**Comment by RedDevilus** [comment]
Besides it's not unheard of, like memcards has its own thing, people don't need central location for the sake of just being central. Relative location can make more sense.

**Comment by Tokman5** [comment]
sorry stenzek

File: pcsx2-qt/Settings/GraphicsSettingsWidget.cpp [hunk 2]
```diff
@@ -320,12 +320,18 @@ GraphicsSettingsWidget::GraphicsSettingsWidget(SettingsDialog* dialog, QWidget*
...
+		SettingWidgetBinder::BindWidgetToFolderSetting(sif, m_ui.videoDumpingDirectory, m_ui.videoDumpingDirectoryBrowse, m_ui.videoDumpingDirectoryOpen, m_ui.videoDumpingDirectoryReset,
+			"Folders", "Videos", Path::Combine(EmuFolders::DataRoot, "videos"));
+
 		SettingWidgetBinder::BindWidgetToStringSetting(sif, m_ui.videoCaptureContainer, "EmuCore/GS", "VideoCaptureContainer");
 		connect(m_ui.videoCaptureContainer, &QComboBox::currentIndexChanged, this, &GraphicsSettingsWidget::onVideoCaptureContainerChanged);
 
 		SettingWidgetBinder::BindWidgetToIntSetting(
 			sif, m_ui.videoCaptureBitrate, "EmuCore/GS", "VideoCaptureBitrate", Pcsx2Config::GSOptions::DEFAULT_VIDEO_CAPTURE_BITRATE);
 
+		SettingWidgetBinder::BindWidgetToBoolSetting(sif, m_ui.enableVideoCaptureArguments, "EmuCore/GS", "EnableVideoCaptureParameters", false);
```
Comment (stenzek) [inline pcsx2-qt/Settings/GraphicsSettingsWidget.cpp:332]: Should we disable the capture arguments if the enable manual option isn't changed? (i.e. connect to `stateChanged`, and enable/disable `videoCaptureArguments`). See audio options for an example.

**Review by stenzek — COMMENTED** [review]
Otherwise LGTM

**Comment by stenzek** [comment]
It's fine, I'm probably the one who should've been reviewing it anyway :)

File: pcsx2-qt/Settings/GraphicsSettingsWidget.cpp (hunk 2 above)
Comment (Tokman5) [reply pcsx2-qt/Settings/GraphicsSettingsWidget.cpp:332]: That is what I was wondering whether to do.
Done.

**Review by Mrlinkwii — APPROVED** [review]

**Review by JordanTheToaster — APPROVED** [review]

**Pull request merged**